src/shared/tools.h

```diff
@@ -4,6 +4,7 @@
 #define _TOOLS_H
 #include <utility>
 #include <vector>
+#include <random>
```

Comment:
We should probably *not* do this. @robalni has had a very good point, which is STL headers likely slowing down compilation. I think a `random.h` is more appropriate for these new declarations.